File: frontend/src/services/staffService.ts
import { apiClient } from '../api/client';
import { ApiResponse, CreateStaffRequest, StaffResponse } from '../types';

export const staffService = {
  create: async (data: CreateStaffRequest): Promise<ApiResponse<StaffResponse>> => {
    const response = await apiClient.post<ApiResponse<StaffResponse>>('/api/v1/staff', data);
    return response.data;
  },

  getAll: async (): Promise<ApiResponse<StaffResponse[]>> => {
    const response = await apiClient.get<ApiResponse<StaffResponse[]>>('/api/v1/staff');
    return response.data;
  },

  getById: async (id: number): Promise<ApiResponse<StaffResponse>> => {
    const response = await apiClient.get<ApiResponse<StaffResponse>>(`/api/v1/staff/${id}`);
    return response.data;
  },

  update: async (id: number, data: CreateStaffRequest): Promise<ApiResponse<StaffResponse>> => {
    const response = await apiClient.put<ApiResponse<StaffResponse>>(`/api/v1/staff/${id}`, data);
    return response.data;
  },

  delete: async (id: number): Promise<ApiResponse<string>> => {
    const response = await apiClient.delete<ApiResponse<string>>(`/api/v1/staff/${id}`);
    return response.data;
  },
};

File: backend/src/main/java/com/csms/staff/service/StaffService.java
package com.csms.staff.service;

import com.csms.common.response.ApiResponse;
import com.csms.staff.dto.request.CreateStaffRequest;
import com.csms.staff.dto.response.StaffResponse;

import java.util.List;

public interface StaffService {

    ApiResponse<StaffResponse> create(CreateStaffRequest request);

    ApiResponse<List<StaffResponse>> getAll();

    ApiResponse<StaffResponse> getById(Long id);

    ApiResponse<StaffResponse> update(Long id, CreateStaffRequest request);

    ApiResponse<String> delete(Long id);
}